crates/services/consensus_module/poa/src/service_test/trigger_tests.rs

```diff
@@ -194,6 +204,47 @@ async fn interval_trigger_produces_blocks_periodically() -> anyhow::Result<()> {
     Ok(())
 }
 
+// In case of error during block creation, the block should be retried after a delay of block time.
+#[tokio::test(start_paused = true)]
+async fn retry_block_creation_in_case_error() -> anyhow::Result<()> {
```

Comment:
Thank you for the test!=) Often, tests with timeout fail because when we run all tests in parallel, the task scheduler could be busy. 

If we want to verify that we continue to produce blocks even in the case of error, maybe it is better to check that we called `produce_and_execute` method several times instead

Reply:
I have changed the test to be notify based instead of time based. I think we could follow this standard for the others test of POA that are in this file. Do you want me to make a separate issue ?

Reply:
I think we can leave other tests for now as it is